### PR TITLE
Submit/newm

### LINKED ISSUE #180320: Pkg-request: newm
**Project description**
A Wayland compositor that has a deferent take on how tiling wm should be on laptops/trackpads

**Metadata**

* homepage URL: https://github.com/jbuchermn/newm
* source URL: https://github.com/jbuchermn/newm
* license: mit
* platforms: linux

### PR DESCRIPTION
**Pull request by jbuchermn**
###### Motivation for this change

Initially package the Wayland compositor newm and its abstraction layer pywm

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/top-level/all-packages.nix
Comment (jonringer) [inline pkgs/top-level/all-packages.nix:34217]: please try to place these in a more sorted location. Appending to end just causes a lot of conflicts

File: pkgs/development/python-modules/pywm/default.nix
Comment (jonringer) [inline pkgs/development/python-modules/pywm/default.nix:3]: Using `pythonXPackages` isn't allowed from `python-modules` as this will mix interpreter versions for a given package. For example, if you were to do `python37Packages.<mypkg>`, but the expression pulled from `python3[8]Packages`. Python will be mis-importing several libraries, and this will likely cause issues at runtime.

Please refer to [Python dependency handling](https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#handling-dependencies) on the best practices for handling dependencies within `python-modules`. Namely, you should be importing each python package individually, and not refer to another python package set (E.g. `python3Packages`).

If you're new to nixpkgs, you're free to view some of my videos on nixpkgs and python:
- [Python packaging using nixpkgs](https://www.youtube.com/watch?v=jXd-hkP4xnU)
- [Adding a package to nixpkgs unstable](https://www.youtube.com/watch?v=fvj8H5yUKu8)

File: pkgs/top-level/all-packages.nix
Comment (jbuchermn) [reply pkgs/top-level/all-packages.nix]: Thank you for your review; I adjusted the placing of both packages

File: pkgs/development/python-modules/pywm/default.nix
Comment (jbuchermn) [reply pkgs/development/python-modules/pywm/default.nix]: I tried to adjust the import handling accordingly (also looking at other python packages)

**Comment by Ashe** [comment]
Is there anything holding this back? I've been keeping my eye on this for a while now, can't wait to see it get submitted :)

File: maintainers/maintainer-list.nix
Comment (AndersonTorres) [inline maintainers/maintainer-list.nix:5383]: This file should be in a separate commit with the title `maintainers: add myself`

File: pkgs/applications/window-managers/newm/default.nix
Comment (AndersonTorres) [inline pkgs/applications/window-managers/newm/default.nix:2]: Each line should start with comma, not end with it.

```suggestion
{ lib
, fetchFromGitHub
```

And do the same below

File: pkgs/applications/window-managers/newm/default.nix
Comment (AndersonTorres) [inline pkgs/applications/window-managers/newm/default.nix:4]:
```suggestion
```

File: pkgs/applications/window-managers/newm/default.nix
Comment (AndersonTorres) [inline pkgs/applications/window-managers/newm/default.nix:26]: Why propagate it?

File: pkgs/applications/window-managers/newm/default.nix
Comment (AndersonTorres) [inline pkgs/applications/window-managers/newm/default.nix:37]: Shouldn't it be `dontCheck = true;`?

File: pkgs/applications/window-managers/newm/default.nix
Comment (AndersonTorres) [inline pkgs/applications/window-managers/newm/default.nix:45]:
```suggestion
    description = "Tiling Wayland compositor written with touchpads and laptops in mind";
```

File: pkgs/development/python-modules/pywm/default.nix
Comment (AndersonTorres) [inline pkgs/development/python-modules/pywm/default.nix:5]: Why the specific meson?

File: pkgs/development/python-modules/pywm/default.nix
Comment (AndersonTorres) [inline pkgs/development/python-modules/pywm/default.nix:2]: Don't bring the world inside the expression. Pick the packages you need explicitly.

File: pkgs/development/python-modules/pywm/default.nix
Comment (AndersonTorres) [inline pkgs/development/python-modules/pywm/default.nix:18]: Bring the specific xorg packages you need, instead of bringing it all.

File: pkgs/development/python-modules/pywm/default.nix
Comment (AndersonTorres) [inline pkgs/development/python-modules/pywm/default.nix:23]: Why such whitespaces?

Also, put those inputs in alphabetical order.

File: pkgs/development/python-modules/pywm/default.nix
Comment (AndersonTorres) [inline pkgs/development/python-modules/pywm/default.nix:39]: If the source repo uses tags, bring the specific tag instead of a opaque hash string.

```suggestion
    rev = "v${version}";
```

If the source repo does not use tags, then why did you set the `version` above?

File: pkgs/development/python-modules/pywm/default.nix
Comment (AndersonTorres) [inline pkgs/development/python-modules/pywm/default.nix:68]: Unneeded comment. Also, use `mesa` directly.

File: pkgs/development/python-modules/pywm/default.nix
Comment (AndersonTorres) [inline pkgs/development/python-modules/pywm/default.nix:90]:
```suggestion
    description = "Python based Wayland compositor library based on wlroots";
```

The `longDescription`, when present, should be a longer, more descriptive text.

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

**Comment by jbuchermn** [comment]
Closing this for now - going with a flakes-installation seems like a more straight-forward way.

**Comment by AndersonTorres** [comment]
OK then. I will pick this later.

**Pull request closed without merging**
